query/src/main/scala/filodb/query/exec/RangeVectorTransformer.scala

```diff
@@ -142,12 +142,29 @@ final case class InstantVectorFunctionMapper(function: InstantFunctionId,
   }
 }
 
+private abstract class FuncIterator(rows: Iterator[RowReader])  extends Iterator[RowReader] {
```

Comment:
Have you evaluated impact if you use this abstract class in the RangeVector trait ?

Reply:
no

Reply:
What issues do you expect?

Reply:
I think it is worth a shot. There may be much good out of this. Right now developers can forget to inherit from this interface.

Reply:
I'm just going for the simplest changes for now, avoiding any possible destabilization.